From kristen.c.accardi@intel.com Wed Nov  2 19:44:36 2005
Subject: [patch 4/4] pci: call pci_read_irq for bridges
From: Kristen Accardi <kristen.c.accardi@intel.com>
To: <rajesh.shah@intel.com>, <greg@kroah.com>, <len.brown@intel.com>
Date: Wed, 02 Nov 2005 16:55:49 -0800
Message-Id: <1130979349.8321.50.camel@whizzy>

Call pci_read_irq() for bridges too, so that the pin value
is stored for bridges that require interrupts.

Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/probe.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/pci/probe.c
+++ gregkh-2.6/drivers/pci/probe.c
@@ -625,6 +625,7 @@ static int pci_setup_device(struct pci_d
 		/* The PCI-to-PCI bridge spec requires that subtractive
 		   decoding (i.e. transparent) bridge must have programming
 		   interface code of 0x01. */ 
+		pci_read_irq(dev);
 		dev->transparent = ((dev->class & 0xff) == 1);
 		pci_read_bases(dev, 2, PCI_ROM_ADDRESS1);
 		break;
